### DISCUSSION TITLE
Are vvenc motion vectors only-distortion optimized or rate-distortion optimized?

### DISCUSSION THREAD

**Discussion by cheptil** [fraunhoferhhi/vvenc#344] [General]
Hello to all

As soon as standard is only define the decoder side, there are many possibilities to implement motion estimation (ME) algorithm on encoder side

So, my question is: does ME in vvenc doing only-distortion optimization or rate-distortion optimization for finding motion vectors (MV)?

If it does rate-distortion optimization, is it mean that codec make full pass after motion estimation stage for calculate rate-distortion cost for current MV?

**Comment by adamjw24** [comment]
> So, my question is: does ME in vvenc doing only-distortion optimization or rate-distortion optimization for finding motion vectors (MV)?

VVenC is optimized for maximum quality, so yes, it does an RD-optimization in motion search. Of course its more complicated than that, starting with a rough SAD only search, and later refining using SATD, while all that time also accounting for MVD signaling bits.

> If it does rate-distortion optimization, is it mean that codec make full pass after motion estimation stage for calculate rate-distortion cost for current MV?

A full residual coding pass is only done for a single optimal MV candidate. Of course a lot more candidates are tested in all the different merge  modes VVC offers, but those are independent of motion search.

**Reply by cheptil** [reply]
Hello, @adamjw24

Thank you very much for your reply!

Am I understand right, that if there are two ME answers and first of them better in ME RDO meaning than second it can be so that second better in full pass RDO meaning than first?
